test/e2e/lib/flows/login-flow.js

```diff
@@ -95,6 +96,13 @@ export default class LoginFlow {
 		}
 
 		await loginPage.login( this.account.email || this.account.username, this.account.password );
+
+		if ( process.env.FLAGS === 'nav-unification' ) {
+			// Makes sure that the nav-unification welcome modal will be dismissed.
+			const guideComponent = new GuideComponent( this.driver, 1000 );
+			await guideComponent.dismissModal( '.nav-unification-modal' );
```

Comment:
Should we consider why we need to pass a selector here? If we've init'd a component with a selector should that not be enough for it to know sensible defaults about what elements to expect? I like the idea of having an _option_ to customise the selector but it shouldn't be required.

We could use the `this.expectedElementSelector` as the root context for selecting the correct elements to dismiss the guide.

Reply:
This is answered here https://github.com/Automattic/wp-calypso/pull/50195#discussion_r578489363